src/Tools/Source/CompilerGeneratorTools/Source/CSharpErrorFactsGenerator/ErrorGenerator.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// We only build the Source Generator in the netstandard target
+#if NETSTANDARD
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Internal.CSharpErrorFactsGenerator
+{
+    [Generator]
+    public sealed partial class ErrorGenerator : IIncrementalGenerator
+    {
+        private readonly struct Model : IEquatable<Model>
+        {
+            public ImmutableArray<string> ErrorNames { get; }
+
+            public Model(ImmutableArray<string> errorNames)
+            {
+                ErrorNames = errorNames;
+            }
+
+            public override bool Equals(object? obj)
+            {
+                return obj is Model model && Equals(model);
+            }
+
+            public bool Equals(Model other)
+            {
+                if (ErrorNames.Length != other.ErrorNames.Length)
+                {
+                    return false;
+                }
+
+                for (int i = 0; i < ErrorNames.Length; i++)
+                {
+                    if (!ErrorNames[i].Equals(other.ErrorNames[i], StringComparison.Ordinal))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            public override int GetHashCode()
+            {
+                return -787298253 + ErrorNames.GetHashCode();
+            }
+
+            public static bool operator ==(Model left, Model right)
+            {
+                return left.Equals(right);
+            }
+
+            public static bool operator !=(Model left, Model right)
+            {
+                return !(left == right);
+            }
```

Comment:
needed?

Reply:
I don't know how the compiler will compare the `Model` to determine whether it needs to re-run or not. So I fully implemented equality and hash code.